packages/x-components/src/x-modules/search/store/actions/set-query-from-url.action.ts

```diff
@@ -0,0 +1,11 @@
+import { SearchXStoreModule } from '../types';
+
+export const setQueryFromUrl: SearchXStoreModule['actions']['setQueryFromUrl'] = (
```

Comment:
Wait is this the same exact code than the other action in the search-box module? Is there a possibility to move this function somewhere else and just use it once in the two modules instead of having to copies?

Reply:
After a couple of ideas, I didn't like none of them. 

- If you create a helper method you need to pass around the commit and the query, and the types are a mess.
- If you create a generic helper method like `export const setQueryFromUrl: T['actions']['setQueryFromUrl'] = <T>(...)` it doesn't work either.

We can take a look together but I think that these actions should keep separate as in the future they could have different implementations